pkg/yurtadm/util/yurthub/yurthub.go

```diff
@@ -170,3 +177,31 @@ func CleanHubBootstrapConfig() error {
 	}
 	return nil
 }
+
+// useRealServerAddr check if the server-addr from yurthubTemplate is default value: 127.0.0.1:6443
+// if yes, we should use the real server addr
+func useRealServerAddr(yurthubTemplate *string, kubernetesServerAddrs string) error {
+	var podObj map[string]interface{}
+	err := yaml.Unmarshal([]byte(*yurthubTemplate), &podObj)
+	if err != nil {
+		klog.Errorf("Failed to parse YAML: %v", err)
+	}
+
+	containers := podObj["spec"].(map[string]interface{})["containers"].([]interface{})
+	command := containers[0].(map[string]interface{})["command"].([]interface{})
+	for i, v := range command {
+		if strings.Contains(v.(string), fmt.Sprintf("--server-addr=%s", constants.DefaultServerAddr)) {
+			command[i] = fmt.Sprintf("--server-addr=%s", kubernetesServerAddrs)
+			break
+		}
+	}
+
+	// Convert the modified podObj back to YAML
+	modifiedYaml, err := yaml.Marshal(podObj)
+	if err != nil {
+		return fmt.Errorf("Failed to marshal modified YAML: %v", err)
+	}
+
+	*yurthubTemplate = string(modifiedYaml)
```

Comment:
I think it's not a good idea to use yaml marshal to replace server address, maybe you can reference the following code:
```
	scanner := bufio.NewScanner(bytes.NewReader(fileContent))
	var buffer bytes.Buffer
	for scanner.Scan() {
		line := scanner.Text()
		if strings.Contains(line, targetURL) {
			line = strings.Replace(line, targetURL, replacementURL, -1)
		}
		buffer.WriteString(line + "\n")
	}

	if err := scanner.Err(); err != nil {
		klog.Infof("Error scanning file: %v\n", err)
		return
	}
```